pkg/lifecycle/version_rules.go

```diff
@@ -0,0 +1,195 @@
+//
+// See CheckChartVersionForLifecycle() for more details.
+func (v *VersionRules) getMinMaxVersionInts() error {
+	// e.g: 2.9 - 0.2 = 2.7
+	minVersionStr := v.Rules[(branchVersionMinorSum(v.BranchVersion, -2))].Min
+	maxVersionStr := v.Rules[v.BranchVersion].Max
+
+	var err error
+	var min, max int = 0, 0
+
+	if minVersionStr != "" {
+		min, err = strconv.Atoi(strings.Split(minVersionStr, ".")[0])
+		if err != nil {
+			return err
+		}
+	}
+	if maxVersionStr != "" {
+		max, err = strconv.Atoi(strings.Split(maxVersionStr, ".")[0])
+		if err != nil {
+			return err
+		}
+	}
+
+	v.MinVersion = min
+	v.MaxVersion = max
+	return nil
+}
+
+// convertBranchVersion will convert the received string flag into a float32
+func convertBranchVersion(branchVersion string) (float32, error) {
+	floatVersion, err := strconv.ParseFloat(branchVersion, 32)
+	if err != nil {
+		return 0.0, err
+	}
+	return float32(floatVersion), nil
+}
+
+// ExtractBranchVersion will extract the branch version from the branch name
+func ExtractBranchVersion(branch string) string {
+	parts := strings.Split(branch, "-v")
+	return parts[len(parts)-1]
+}
+
+// CheckChartVersionForLifecycle will
+// Check if the chart version is within the range of the current version:
+//
+//	If the chart version is within the range, return true, otherwise return false
+func (v *VersionRules) CheckChartVersionForLifecycle(chartVersion string) bool {
+	chartVersionInt, _ := strconv.Atoi(strings.Split(chartVersion, ".")[0])
+	/**
+	Rule Example:
+	Branch version: 2.9
+	Min version: 104
+	Max version: 105
+	Therefore, the chart version must be >= (104 - 2) and < 105
+	i.e: 102 <= chartVersion < 105
+	*/
+	if chartVersionInt >= v.MinVersion && chartVersionInt < v.MaxVersion {
+		return true
+	}
+	return false
+}
+
+// CheckChartVersionToRelease will return if the current versyion being analyzed is the one to be released or not
+func (v *VersionRules) CheckChartVersionToRelease(chartVersion string) (bool, error) {
+	chartVersionInt, err := strconv.Atoi(strings.Split(chartVersion, ".")[0])
+	if err != nil {
+		logrus.Errorf("failed to check version to release for chartVersion:%s with error:%v", chartVersion, err)
+		return false, err
+	}
+	return chartVersionInt == (v.MaxVersion - 1), nil
+}
+
+// CheckForRCVersion checks if the chart version contains the "-rc" string indicating a release candidate version.
+func (v *VersionRules) CheckForRCVersion(chartVersion string) bool {
+	if strings.Contains(strings.ToLower(chartVersion), "-rc") {
+		return true
+	}
+	return false
+}
+
+func branchVersionMinorSum(branchVersion string, number int) string {
+	parts := strings.Split(branchVersion, ".")
+	major, err := strconv.Atoi(parts[0])
+	if err != nil {
+		// Handle error
+		return ""
+	}
+
+	minor, err := strconv.Atoi(parts[1])
+	if err != nil {
+		// Handle error
+		return ""
+	}
+
+	minor += number
+	return fmt.Sprintf("%d.%d", major, minor)
+}
```

Comment:
I believe the semver package that we already use in other parts of our code can be used to handle this

Reply:
I tried this before; it adds more complexity to a simple thing. 
In this case, I prefer to do the raw operations. 
@tashima42

Reply:
👍 I just noticed this is not ecm distro tools